instrumentation/excon/lib/opentelemetry/instrumentation/excon/middlewares/tracer_middleware.rb

```diff
@@ -71,43 +77,35 @@ def self.around_default_stack
           private
 
           def handle_response(datum)
-            if datum.key?(:otel_span)
-              datum[:otel_span].tap do |span|
-                return span if span.end_timestamp
+            datum.delete(:otel_span)&.tap do |span|
+              return span unless span.recording?
 
-                if datum.key?(:response)
-                  response = datum[:response]
-                  span.set_attribute('http.status_code', response[:status])
-                  span.status = OpenTelemetry::Trace::Status.error unless (100..399).include?(response[:status].to_i)
-                end
-
-                span.status = OpenTelemetry::Trace::Status.error("Request has failed: #{datum[:error]}") if datum.key?(:error)
+              if datum.key?(:response)
+                response = datum[:response]
+                span.set_attribute(OpenTelemetry::SemanticConventions::Trace::HTTP_STATUS_CODE, response[:status])
+                span.status = OpenTelemetry::Trace::Status.error unless (100..399).include?(response[:status].to_i)
```

Comment:
Nit: Is there any performance gain if we extracted this range as a Constant?

Reply:
I whipped together a quick benchmark. The answer is it doesn't matter.
I ran the benchmark with and without GC and the results were the same.

## Benchmark
```
require "benchmark"

CONSTANT = (100..399)

n = 10_000_000

GC.disable if ENV.include?("DISABLE_GC")

Benchmark.bmbm do |x|
  x.report("constant") do
    n.times do
      CONSTANT.include?(200) && !CONSTANT.include?(500)
    end
  end

  x.report("inline") do
    n.times do
      (100..399).include?(200) && !(100..399).include?(500)
    end
  end
end
```

## Results
### With GC
```
Rehearsal --------------------------------------------
constant   1.923770   0.025237   1.949007 (  1.989501)
inline     1.867091   0.025007   1.892098 (  1.996068)
----------------------------------- total: 3.841105sec

               user     system      total        real
constant   1.900203   0.027615   1.927818 (  1.980085)
inline     1.850816   0.021425   1.872241 (  1.909520)
```

### Without GC
```
Rehearsal --------------------------------------------
constant   1.870138   0.014792   1.884930 (  1.896819)
inline     1.826463   0.018247   1.844710 (  2.032935)
----------------------------------- total: 3.729640sec

               user     system      total        real
constant   1.851058   0.014311   1.865369 (  1.878788)
inline     1.793849   0.014889   1.808738 (  1.822874)
```

Reply:
Thanks for humoring me and ultimately answering my question :)